cwf/gateway/helm/cwf-kubevirt/templates/bin/_install_gateway.sh.tpl

```diff
@@ -166,6 +166,11 @@ cp sessiond.yml /var/opt/magma/configs/
 cp docker-compose.yml /var/opt/magma/docker/
 cp .env /var/opt/magma/docker/
 
+# Set log options for log aggregation
+if [ {{ .Values.cwf.log_aggregation.enabled }} ]; then
+    sed -i "s/tag: \"{{ "{{" }}\.Name{{ "}}" }}\/{{ "{{" }}\.ID{{ "}}" }}\"/fluentd-async-connect: \"true\"/g" docker-compose.yml
```

Comment:
I know this is gross, but let me do my best to explain why I'm doing this:

Docker won't allow log options for the wrong log driver, so we can't just set options for each driver and then use the env var to determine which one will run. We could instead just remove the options, but this will force every service to be dependent on the td-agent-bit service running. If that service fails to startup, we lose all functionality - this seems like unnecessarily adding a single point of failure.

Docker compose doesn't allow for control flow so we can't if else which log anchor runs.
Docker compose also won't allow you to use env vars like ${LOG OPTION NAME}: ${LOG_OPTION_VALUE}

This leaves us with either templating the entire docker-compose file, or scripting the change I have here.
The docker-compose changes quite often and I think there's a greater risk of the template not staying in sync with master, so I opted for the sed command. Hopefully it'll be something we can set and forget.

Reply:
yeesh, do we have python available to use for a more structured way of manipulating the file here?

What about templating the docker-compose file as part of the Helm chart? You could run `helm template ... > docker-compose.yml` for local development work (and wrap this with a script for convenience).

And finally - how about writing our own log driver? It actually looks mostly straightforward: https://medium.com/@software_factotum/writing-a-docker-log-driver-plugin-7275d99d07be

Reply:
Yeah templating the file is probably the best option here